### PR TITLE
[AIRFLOW-5836] Pin azure-storage-blob version to <12

### PR DESCRIPTION
**Pull request by mik-laj**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5836
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -152,12 +152,13 @@ def write_version(filename: str = os.path.join(*["airflow", "git_version"])):
...
-    'azure-storage>=0.34.0',
-    'azure-mgmt-resource>=2.2.0',
-    'azure-mgmt-datalake-store>=0.5.0',
-    'azure-datalake-store>=0.0.45',
     'azure-cosmos>=3.0.1',
+    'azure-datalake-store>=0.0.45',
     'azure-mgmt-containerinstance>=1.5.0',
+    'azure-mgmt-datalake-store>=0.5.0',
+    'azure-mgmt-resource>=2.2.0',
+    'azure-storage-blob<12.0',
```
Comment (feluelle) [inline setup.py:160]:
```suggestion
    'azure-storage-blob<12.0',  # TODO: Remove this dependency after https://github.com/snowflakedb/snowflake-connector-python/issues/225 is solved and released.
```

**Comment by feluelle** [comment]
I restarted the failed check.

It was failing due to:
```
  Traceback (most recent call last):

    /usr/local/lib/python3.5/unittest/mock.py line 1159 in patched

      return func(*args, **keywargs)

    tests/jobs/test_local_task_job.py line 161 in test_heartbeat_failed_fast

      self.assertGreaterEqual((time2 - time1).total_seconds(), job.heartrate)

   AssertionError: 1.998126 not greater than or equal to 2
```
(another flaky test :/)

File: setup.py (hunk 1 above)
Comment (feluelle) [reply setup.py:160]: @mik-laj what do you think about this?

Comment (mik-laj) [reply setup.py:160]: I agree, but I think we can add this comment when merging to avoid creating another build.

Comment (feluelle) [reply setup.py:160]: Okay. I am fine with it if we have a ticket. I am creating a ticket.

**Review by feluelle — APPROVED** [review]
Since this is a temporarily solution there is the Jira Ticket to remove this extra dependency once the issue is solved. See https://issues.apache.org/jira/browse/AIRFLOW-5837 for more information.

**Pull request merged**